src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c

```diff
@@ -776,6 +776,26 @@ static int xclmgmt_read_subdev_req(struct xclmgmt_dev *lro, char *data_ptr, void
 		xclmgmt_subdev_get_data(lro, subdev_req->offset,
 			subdev_req->size, resp, &current_sz);
 		break;
+	case XCL_SDR_BDINFO:
```

Comment:
Why can not reuse XCL_BDINFO?

Reply:
If XCL_BDINFO used, then need to add a check to redirect the request to either xmc or hwmon_sdm driver.
Please let me know if adding a check is better? Again this check will be like if hwmon_sdm exist ?

Reply:
Will xmc and hwmon_sdm co-exist? It seems not to me.

Reply:
ok, but how to know which callback to be called if I use XCL_BDINFO for both?

For xmc: we have xocl_xmc_get_sensors()

For hwmon_sdm: we have xocl_hwmon_sdm_get_sensors()

Reply:
I think there are 2 options.
1. use the same device id for  xmc and sdm. You may unify the callback table.
2. check if xocl_hwmon_sdm_get_sensors() returns -NODEV, then call xocl_xmc_get_sensors() if it is.

Reply:
i will try it and will update.

Reply:
Hi @houlz0507 , I have added your suggested approach for dealing with BDINFO.

Reply:
@rajkumar-xilinx , could you confirm that the data format returned from xclmgmt for BDINFO is the same as before (XMC)?

Reply:
Hi @maxzhen , response data format is not same between XMC and hwmon_sdm driver.

XMC: it uses struct xcl_board_info as the response format.
hwmon_sdm: it receives raw data received from vmr. We are not defining xcl_board_info structure here, since it is data driven.

Reply:
@rajkumar-xilinx , @houlz0507 , I think we should use different sub-op codes for responses from SDR driver (BDINFO, sensors, etc).

Mailbox protocol is b/w xclmgmt and xocl. They may not be always in sync (say in cloud use case). When we making changes in mailbox_proto.h, we're changing the protocol and we should maintain backward compatibility.

A protocol consists of data structure and op-codes. They tie to each other. You can add new op-code and new data structures or obsolete old op-codes and remove them over time. But you don't want to map same op-code to incompatible data structures, which may confuse the peer when the two sides are out of sync.

By "incompatible data structure", I mean that the data structure that can't be parsed properly by code before the change.

In this PR, the response data structure is not compatible with the one from XMC, so different op-code works better.

Let me know if you think otherwise...

Reply:
Hi @maxzhen thanks for suggestions. I have updated the changes to use separate & new opcodes for SDR driver.